contracts/Marketplace.sol

```diff
@@ -120,11 +130,13 @@ contract Marketplace is Collateral, Proofs, StateRetrieval {
   function markProofAsMissing(SlotId slotId, Period period) public {
     require(slotState(slotId) == SlotState.Filled, "Slot not accepting proofs");
     _markProofAsMissing(slotId, period);
-    address host = getHost(slotId);
+    Slot storage slot = _slots[slotId];
+
     if (missingProofs(slotId) % config.collateral.slashCriterion == 0) {
-      _slash(host, config.collateral.slashPercentage);
+      uint256 slashedAmount = (slot.currentCollateral * config.collateral.slashPercentage) / 100;
+      slot.currentCollateral -= slashedAmount;
```

Comment:
You'd probably want to update `_funds.slashed` as well here

Reply:
As I mentioned in the PR's description, I have removed the collateral funds tracking as part of the Collateral contract removal. The `_funds` in play here are the `MarketplaceFunds` that do not have `slashed` property. Do you want me to incorporate that? Or will this be removed with the fuzzing anyhow?

Reply:
The implementation will be different when #37 (fuzzing) is merged, but I'd still like to check that slashed funds can never be extracted from the contract. Probably the easiest way of doing that now is by keeping track of slashed funds in the Marketplace, since we no longer need the Collateral contract.